src/Orleans/Core/ClusterClient.cs

```diff
@@ -143,31 +164,36 @@ public static ClusterClient Create(FileInfo configFile)
         public static ClusterClient Create(ClientConfiguration configuration) => new ClusterClient(configuration);
 
         /// <inheritdoc />
-        public void Start()
+        public async Task Start()
         {
-            lock (this.initLock)
+            this.ThrowIfDisposed();
+            using (await this.initLock.LockAsync())
```

Comment:
yup, this one is indeed outdated. Nevertheless I see several in OusideRuntimeClient that should be .ConfigureAwait(false) also. If you look at [GrainClient.InternalInitialize](https://github.com/dotnet/orleans/pull/2822/files#diff-9f940665838d22872b8984d950096293L194) it would enqueue the initialization work in a background thread. We are no longer doing that, so we either do it too, or be more careful to not propagate the initialization context, and not do any blocking calls in that thread (but I suppose you already did that).